samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+                    }
+
+                    var errorOutputContainer = CreateTranscriptionEnvironmentVariables.ErrorReportOutputContainer;
+                    await StorageUtilities.WriteTextFileToBlobAsync(CreateTranscriptionEnvironmentVariables.AzureWebJobsStorage, exceptionMessage, errorOutputContainer, errorTxtName, log).ConfigureAwait(false);
+                }
+
+                throw;
+            }
+            catch (FormatException e)
+            {
+                log.LogError($"Failed with FormatException. Write message for job with name {jobName} to report file.");
+
+                var errorTxtName = jobName + "_error.txt";
+                var responseMessage = $"Transcription failed with FormatException:\n{e.Message}";
+
+                var errorOutputContainer = CreateTranscriptionEnvironmentVariables.ErrorReportOutputContainer;
+                await StorageUtilities.WriteTextFileToBlobAsync(CreateTranscriptionEnvironmentVariables.AzureWebJobsStorage, responseMessage, errorOutputContainer, errorTxtName, log).ConfigureAwait(false);
+
+                return string.Empty;
+            }
+            catch (TimeoutException e)
+            {
+                log.LogError($"Timeout while creating post, re-enqueueing transcription create. Message: {e.Message}");
+                var createTranscriptionSBConnectionString = CreateTranscriptionEnvironmentVariables.CreateTranscriptionServiceBusConnectionString;
+
+                foreach (var message in messages)
+                {
+                    ServiceBusUtilities.SendServiceBusMessageAsync(createTranscriptionSBConnectionString, message, log, 2).GetAwaiter().GetResult();
+                }
+
+                return string.Empty;
+            }
+
+            var reenqueueingTimeInSeconds = MapDatasetSizeToReenqueueingTimeInSeconds(CreateTranscriptionEnvironmentVariables.AudioDatasetSize);
+
+            var transcriptionMessage = new PostTranscriptionServiceBusMessage(
+                subscription,
+                locationString,
+                jobName,
+                StartDateTime.ToString(CultureInfo.InvariantCulture),
+                locale,
+                modelIds.Any(),
+                CreateTranscriptionEnvironmentVariables.AddSentimentAnalysis,
+                CreateTranscriptionEnvironmentVariables.AddEntityRedaction,
+                0,
+                reenqueueingTimeInSeconds,
+                reenqueueingTimeInSeconds);
+            var fetchTranscriptionSBConnectionString = CreateTranscriptionEnvironmentVariables.FetchTranscriptionServiceBusConnectionString;
+            log.LogInformation($"FetchTranscriptionServiceBusConnectionString from settings: {fetchTranscriptionSBConnectionString}");
+
+            ServiceBusUtilities.SendServiceBusMessageAsync(fetchTranscriptionSBConnectionString, transcriptionMessage.CreateMessageString(), log, reenqueueingTimeInSeconds).GetAwaiter().GetResult();
```

Comment:
What happen if SendServiceBusMessageAsync() fails? Will this request get lost?

Reply:
yes, but I don't see any way around that. We don't want to keep our Azure Function running forever, but we also need to re-schedule the running. Do you think that it's likely that sending a message to service bus will fail?